clinica/iotools/converter_utils.py

```diff
@@ -76,149 +185,131 @@ def print_statistics(summary_file, num_subjs, ses_aval, mmt):
+        summary += f"{ses}\n{compute_table(mod_dict)}\n\n"
+    return summary
 
 
-def print_table(summary_file, double_dict):
+def compute_table(mod_dict: dict) -> str:
+    """Builds a table, encoded as a string, describing the
+    modalities in the given dictionary.
 
-    # Find all keys at the second level of the dictionary
-    diagnoses = set()
-    mods = double_dict.keys()
-    for mod in mods:
-        diagnoses = diagnoses | set(double_dict[mod].keys())
-
-    diagnoses = list(diagnoses)
-    diagnoses.sort()
-
-    summary_file.write("\t")
-    for diag in diagnoses:
-        summary_file.write(f"\t| {diag}")
-    summary_file.write("\n" + "-" * 8 * (len(diagnoses) + 2) + "\n")
-
-    for mod in double_dict.keys():
-        summary_file.write(f"{mod}")
-        if len(mod) < 8:
-            summary_file.write("\t")
-        for diag in diagnoses:
-            if diag not in double_dict[mod]:
-                summary_file.write("\t| 0")
-            else:
-                summary_file.write(f"\t| {double_dict[mod][diag]}")
-        summary_file.write("\n")
-
-
-def has_one_index(index_list):
-    if len(index_list) == 1:
-        return index_list[0]
-    if len(index_list) == 0:
-        return -1
-    if len(index_list) > 1:
-        raise ValueError("Multiple indexes found")
-
-
-class MissingModsTracker:
-    """Class used for tracking the number of missing modalities in a database."""
-
-    def __init__(self, ses, mod_list=None):
-        self.missing = {}
-        self.ses = ses
-        if mod_list:
-            for s in ses:
-                self.missing.update({s: {"session": 0}})
-                for mod in mod_list:
-                    self.missing[s].update({mod: 0})
-        else:
-            for s in ses:
-                self.missing.update(
-                    {
-                        s: {
-                            "session": 0,
-                            "dwi": 0,
-                            "func": 0,
-                            "fieldmap": 0,
-                            "flair": 0,
-                            "t1w": 0,
-                        }
-                    }
-                )
-
-    def add_missing_mod(self, ses, mod):
-        """Increase the number of missing files for the input modality.
-
-        Args:
-            ses: name of the session
-            mod: modality missing
-        """
-        self.missing[ses][mod] += 1
-
-    def increase_missing_ses(self, ses):
-        self.missing[ses]["session"] += 1
-
-    def get_missing_list(self):
-        """Return the hash map mods_missing.
+    Parameters
+    ----------
+    mod_dict : dict
+        Dictionary of modalities for which to compute the table.
 
-        Returns:
-             The hash map containing the list of missing files.
-        """
-        return self.missing
+    Returns
+    -------
+    table : str
+        The summary table as a string.
+    """
+    diagnoses = sorted(list(set().union(*mod_dict.values())))
```

Comment:
`sorted` works on sets directly:

```suggestion
    diagnoses = sorted(set().union(*mod_dict.values()))
```

Reply:
Thanks ! 🙏